src/main/java/com/mongodb/spark/sql/connector/schema/ConverterHelper.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.mongodb.spark.sql.connector.schema;
+
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Base64;
+
+import org.bson.BsonValue;
+import org.bson.codecs.BsonValueCodec;
+import org.bson.codecs.Codec;
+import org.bson.json.JsonMode;
+import org.bson.json.JsonWriterSettings;
+
+interface ConverterHelper {
+  Codec<BsonValue> BSON_VALUE_CODEC = new BsonValueCodec();
+  JsonWriterSettings DEFAULT_JSON_WRITER_SETTINGS =
+      JsonWriterSettings.builder()
+          .outputMode(JsonMode.RELAXED)
+          .binaryConverter(
+              (value, writer) ->
+                  writer.writeString(Base64.getEncoder().encodeToString(value.getData())))
+          .dateTimeConverter(
+              (value, writer) -> {
+                ZonedDateTime zonedDateTime = Instant.ofEpochMilli(value).atZone(ZoneOffset.UTC);
+                writer.writeString(DateTimeFormatter.ISO_DATE_TIME.format(zonedDateTime));
+              })
+          .decimal128Converter((value, writer) -> writer.writeString(value.toString()))
+          .objectIdConverter((value, writer) -> writer.writeString(value.toHexString()))
+          .symbolConverter((value, writer) -> writer.writeString(value))
+          .build();
+
+  static Codec<BsonValue> getBsonValueCodec() {
```

Comment:
We should convert `ConverterHelper` to a `final` class with a private constructor for the following reasons:

1. The ability to define `static` methods in interfaces was added to Java because it [_"makes it easier for you to organize helper methods in your libraries; you can keep static methods specific to an interface in the same interface rather than in a separate class"_](https://docs.oracle.com/javase/tutorial/java/IandI/defaultmethods.html#static). However, this is not applicable to the interface `ConverterHelper`: the interface is meant to be solely a collection of `static` methods and is not meant to define a type.
2. The purpose of `static` methods `getBsonValueCodec`/`getDefaultJsonWriterSettings` appear to be to provide an identity-agnostic way of getting a BSON value codec / default JSON writer settings. However, the fact that the `static` fields `BSON_VALUE_CODEC`/`DEFAULT_JSON_WRITER_SETTINGS` are also `public`, diminishes the benefit of these methods, because a user can still access the BSON value codec / default JSON writer settings in a way that then allows to rely on their identity.

Generally speaking, an interface may be used only if it is meant to define a type, i.e., to be implemented.

Reply:
I like it better as you suggested - thanks @stIncMale